File: targets/jsts-web/src/vulns/vuln17_cookie.ts
import { Router } from "express";
export const router = Router();
router.get("/vuln17/login", (req, res) => {
  const token = Math.random().toString(16).slice(2);
  // VULN:VULN-17:CWE-614:config auth cookie set without HttpOnly/Secure flags
  res.cookie("auth", token, { httpOnly: false, secure: false });
  res.json({ status: "ok", token });
});

File: targets/python-web/vulns/vuln17_cookie.py
import random
from flask import Blueprint, request, make_response
bp = Blueprint("vuln17", __name__)

@bp.get("/vuln17/login")
def login():
    user = request.args.get("user", "guest")
    token = "%016x" % random.Random(hash(user)).getrandbits(64)
    resp = make_response({"status": "ok", "token": token})
    # VULN:VULN-17:CWE-614:config auth cookie set without HttpOnly/Secure flags
    resp.set_cookie("auth", token, httponly=False, secure=False)
    return resp
